cdisc_rules_engine/operations/parent_library_model_column_order.py

```diff
@@ -0,0 +1,43 @@
+from cdisc_rules_engine.operations.library_model_column_order import (
+    LibraryModelColumnOrder,
+)
+from pandas import Series
+from collections import defaultdict
+
+
+class ParentLibraryModelColumnOrder(LibraryModelColumnOrder):
+    def _execute_operation(self):
+        """
+        Fetches column order for a supp's parent domain from the CDISC library.
+        Returns it as a Series of lists like:
+        0    ["STUDYID", "DOMAIN", ...]
+        1    ["STUDYID", "DOMAIN", ...]
+        2    ["STUDYID", "DOMAIN", ...]
+        ...
+
+        Length of Series is equal to the length of given dataframe.
+        The lists with column names are sorted
+        in accordance to "ordinal" key of library metadata.
+        """
+        domain_to_datasets = self._get_domain_to_datasets()
+        rdomain_names_list = {}
+        return Series(
```

Comment:
Sorry I didn't mention this before, but what happens if the series returned is shorter than the dataset being evaluated?

Reply:
I don't think this can happen unless RDOMAIN column is missing, so I added logic to handle that